OpenSim/Region/CoreModules/Scripting/LSLHttp/UrlModule.cs

```diff
@@ -518,11 +518,16 @@ public void AsyncHttpRequest(IHttpServer server, string path, OSHttpRequest http
                     }
                 }
 
+                // Grab the raw unprocessed original query string, if any.
+                int rawQueryPos = httpRequest.Url.Query.IndexOf('?');
+                string rawQueryStr = (rawQueryPos < 0) ? httpRequest.Url.Query : httpRequest.Url.Query.Substring(rawQueryPos + 1);
+
                 //if this machine is behind DNAT/port forwarding, currently this is being
                 //set to address of port forwarding router
                 requestData.headers["x-remote-ip"] = httpRequest.RemoteIPEndPoint.ToString();
                 requestData.headers["x-path-info"] = pathInfo;
-                requestData.headers["x-query-string"] = queryString;
+                requestData.headers["x-query-string"] = rawQueryStr;        // raw original (SL-compatible)
+                requestData.headers["x-query-string-compat"] = queryString; // processed (old Halcyon scripts)
```

Comment:
If they are old then they'll not be calling the compat header and will break anyway.  The only reason I can thing to introduce this new header is to provide the old method for new scripts being written to use the old method - not sure there's any need for that, especially since the semantics have changed: the processing now skips nulls.